torchopt/pytree.py

```diff
@@ -48,10 +64,80 @@ def tree_flatten_as_tuple(
     return tuple(leaves), treespec
 
 
+def acc_add(*args: T) -> T:
+    """Accumulate addition."""
+    return functools.reduce(operator.add, args)
+
+
+def acc_mul(*args: T) -> T:
+    """Accumulate multiplication."""
+    return functools.reduce(operator.mul, args)
+
+
+def acc_matmul(*args: T) -> T:
+    """Accumulate matrix multiplication."""
+    return functools.reduce(operator.matmul, args)
+
+
+def tree_add_scalar_mul(tree_x: TensorTree, tree_y: TensorTree, alpha: float = 1.0) -> TensorTree:
+    """Computes tree_x + alpha * tree_y."""
+    return tree_map(lambda x, y: x.add(y, alpha=alpha), tree_x, tree_y)
```

Comment:
```suggestion
def tree_add_scalar_mul(tree_x: TensorTree, tree_y: TensorTree, alpha: Optional[float] = None) -> TensorTree:
    """Computes tree_x + alpha * tree_y."""
    if alpha is None:
        return tree_map(lambda x, y: x.add(y), tree_x, tree_y)
    return tree_map(lambda x, y: x.add(y, alpha=alpha), tree_x, tree_y)
```

And reorder functions to match `__all__`.